Jint/ParseOptions.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Jint
+{
+    public interface IParseOptions
+    {
+        /// <summary>
+        /// Gets or sets whether to create compiled <see cref="Regex"/> instances when adapting regular expressions.
+        /// Defaults to <see langword="null"/>, which means that in the case of non-prepared scripts and modules
+        /// regular expressions will be interpreted, otherwise they will be compiled.
+        /// </summary>
+        bool? CompileRegex { get; init; }
+
+        /// <summary>
+        /// Gets or sets the default timeout for created <see cref="Regex"/> instances.
+        /// Defaults to <see langword="null"/>, which means that in the case of non-prepared scripts and modules
+        /// the <see cref="Options.ConstraintOptions.RegexTimeout"/> setting should apply,
+        /// otherwise the default of the <see cref="ParserOptions.RegexTimeout"/> setting (10 seconds).
+        /// </summary>
+        /// <remarks>
+        /// Please note that <see cref="Options.ConstraintOptions.RegexTimeout"/> setting will be ignored
+        /// if this option is set to a value other than <see langword="null"/>.
+        /// </remarks>
+        TimeSpan? RegexTimeout { get; init; }
+
+        /// <summary>
+        /// Gets or sets whether to parse the source code in tolerant mode.
+        /// Defaults to <see langword="true"/>.
+        /// </summary>
+        /// <remarks>
+        /// Please note that there is a plan to change the default to <see langword="false"/> in the next major version.
+        /// </remarks>
+        bool Tolerant { get; init; }
```

Comment:
live below (came back to this), should we just initialize with `true`?

Reply:
I'm not sure I get this. This is an interface, we can't initialize this here, can't we? It's initialized here: https://github.com/sebastienros/jint/pull/1831/files#diff-5196d539da48e2bc4acd76456adfff1ca098ad422ce71e90849e98258c066633R62

Reply:
Lazy eyes, sorry.